lib/active_merchant/billing/gateways/firstdata_e4.rb

```diff
@@ -263,7 +264,7 @@ def commit(action, request, credit_card = nil)
 
         Response.new(successful?(response), message_from(response), response,
           :test => test?,
-          :authorization => response_authorization(action, response, credit_card),
+          :authorization => successful?(response) ? response_authorization(action, response, credit_card) : nil,
```

Comment:
I'm a little concerned about this change. When `#response_authorization` falls through to `#authorization_from`,with what appears to be an unsuccessful transaction, it [returns an empty string](https://github.com/jcbantuelle/active_merchant/blob/master/lib/active_merchant/billing/gateways/firstdata_e4.rb#L292-L294) not `nil`, so I'm worried that this change in behaviour may break some integrations that expect to always have a string here.